From 66d025efdbd43d322b1e4924d03c129dbc90f857 Mon Sep 17 00:00:00 2001
From: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Date: Tue, 12 Jun 2018 18:51:28 -0500
Subject: [PATCH 1214/1240] acpi: blacklist: remove quirk for Dell XPS13 when
 SOF is enabled

Audio works well in I2S mode with SOF (Sound Open Firmware), no
need for this quirk

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
---
 drivers/acpi/blacklist.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/acpi/blacklist.c b/drivers/acpi/blacklist.c
index 995c4d8..edf821b 100644
--- a/drivers/acpi/blacklist.c
+++ b/drivers/acpi/blacklist.c
@@ -91,6 +91,7 @@ static int __init dmi_enable_rev_override(const struct dmi_system_id *d)
 
 static const struct dmi_system_id acpi_rev_dmi_table[] __initconst = {
 #ifdef CONFIG_ACPI_REV_OVERRIDE_POSSIBLE
+#if !IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL)
 	/*
 	 * DELL XPS 13 (2015) switches sound between HDA and I2S
 	 * depending on the ACPI _REV callback. If userspace supports
@@ -105,6 +106,7 @@ static const struct dmi_system_id acpi_rev_dmi_table[] __initconst = {
 		      DMI_MATCH(DMI_PRODUCT_NAME, "XPS 13 9343"),
 		},
 	},
+#endif
 	{
 	 .callback = dmi_enable_rev_override,
 	 .ident = "DELL Precision 5520",
-- 
2.7.4

